src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,96 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { getPostComments, deleteComment, addComment } from '../../api/api';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
+interface Props {
+  post: Post,
+}
 
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comment, setComment] = useState<Comment[] | []>([]);
```

Comment:
```suggestion
  const [comments, setComments] = useState<Comment[] | []>([]);
```
Если там массив, то должно быть множественное число